### ISSUE TITLE
Wabbajack

### ISSUE THREAD

**Issue by kleshas** [rockerbacon/modorganizer2-linux-installer#93]
Do you have any interest in getting wabbajack working in this project?  It's a modlist auto-downloader, runs on win10 exclusively from what I gather.

I tried myself by creating a new run.sh, configuring the SSE game in lutris to run that instead, and the only change inside that .sh is pointing to Wabbajack.exe instead of ModOrganizer.exe.  It runs, downloads its latest version, but then crashes with the following excerpt:

Unhandled exception. System.Windows.Markup.XamlParseException: The invocation of the constructor on type 'Wabbajack.MainWindow' that matches the specified binding constraints threw an exception.
 ---> System.TypeInitializationException: The type initializer for '<Module>' threw an exception.
 ---> System.Runtime.InteropServices.SEHException (0x80004005): External component has thrown an exception.
   at Wabbajack.Lib.LibCefHelpers.Helpers.Init()
   at Wabbajack.Lib.LibCefHelpers.ModuleInitializer.Initialize() in c:\agent\_work\1\s\Wabbajack.Lib\LibCefHelpers\Init.cs:line 127
   at .cctor()

I assume it's just a prereq missing, but according to Wabbajack's devs, there aren't any beyond windows10 itself.

**Comment by rockerbacon** [comment]
> but according to Wabbajack's devs, there aren't any beyond windows10 itself

A quick look at their GitHub suggests the exact opposite. They seem to use a good amount of third party libraries from NuGet.

Considering they use the Microsoft Stack exclusively, I supposed just adding an updated .NET from Winetricks is enough. [This pre-release](https://github.com/rockerbacon/lutris-skyrimse-installers/releases/tag/2.8.0) will do that, see if that is enough to get your script working.

**Comment by kleshas** [comment]
I ran the new .xml, it completed fine, MO2 runs from run.sh.  So I edited the run.sh to Wabbajack.exe from ModOrganizer.exe which I put in the same folder.  Errored out again, after it updates and does some other windows-dll type stuff (creates a 2.0.9.4 folder with a bunch of language files and dlls).
[error.txt](https://github.com/rockerbacon/lutris-skyrimse-installers/files/4848629/error.txt)

**Comment by rockerbacon** [comment]
The proton-launcher is not finding your Steam runtime (line 13 in the logs). What is the output for `$ ls -l "$(readlink -f "$HOME/.steam/root")`?

**Comment by kleshas** [comment]
ls: cannot access '/home/bhava/.local/share/Steam'$'\n\n\n\n''ls -l /home/bhava/.local/share/Steam': No such file or directory

**Comment by rockerbacon** [comment]
There are missing symbolic links in your Steam directory. What distro are you using?

**Comment by kleshas** [comment]
arch

**Comment by rockerbacon** [comment]
Looks like Steam will create a generic name symlink on Ubuntu based distros but will directly use the directory with the Ubuntu name in other distros. [2.7.5](https://github.com/rockerbacon/lutris-skyrimse-installers/releases/tag/2.7.5) should fix the runtime problem. Now I would like to confirm what exactly is necessary to get Wabbajack running.

If you would be willing to help, I'd like you to see if you can run it by doing the following:
1. Backup the current prefix:
```bash
$ mv "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830" "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830_bak"
```
2. Launch Skyrim Special Edition on Steam to force it to create a new prefix
3. Install [2.7.5](https://github.com/rockerbacon/lutris-skyrimse-installers/releases/tag/2.7.5)
4. See if Wabbajack works

If Wabbajack works, it means the .NET version provided by Steam is enough and you can safely delete your backup:
```bash
$ rm -rf "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830_bak"
```

If it doesn't, it might need the updated .NET that was installed earlier. Restore the backup and try to run Wabbajack again:
```bash
$ rm -rf "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830" && mv "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830_bak" "/mnt/veracrypt3/SteamLibrary/steamapps/compatdata/489830"
```

**Comment by kleshas** [comment]
New installer - it 'crashed' just before the 'extracting modorganizer' stage.  I edited the run.sh to point to Wabbajack.exe and it ran - there are graphical glitches - window flickers between black and showing the screen - if I keep the cursor away from the app screen, I can see the whole app, when I cursor into the app window, I only see the thing I am pointing at, like a checkbox, or the settings icons and nothing else.

**Comment by kleshas** [comment]
Error while installing winetricks. (can't move /home/kleshas/.cache/lutris/installer/mod-organizer-2/successful-prefix-preparation, it doesn't exist)  - sha256sum mismatch! Rename /home/kleshas/.cache/winetricks/vcrun2019/vc_redist.x86.exe and try again.

**Comment by rockerbacon** [comment]
That's a known issue with Winetricks. Check #61

**Comment by kleshas** [comment]
Yeah, I remembered that vc issue just after I commented.
OK, 1-4 didnt work.
So I run the restore and ran again - this time it works.. kinda.  The app is running, but black screen unless I move the cursor around - sometimes the whole interface is fine, sometimes just able to see a widget, like the settings button.  Clicking on things like the settings button DOES work, but the screen is black 75% of the time, and you have to move the cursor around to see what's going on.

**Comment by rockerbacon** [comment]
Seeing as the original problem of it not working is fixed, as of [2.8.1](https://github.com/rockerbacon/lutris-skyrimse-installers/releases/tag/2.8.1), I consider this issue to be solved.

The black screen problem will be dealt with in the separate issue above.

**Comment by rockerbacon** [comment]
This partial support will be removed in the next release because adding `dotnet48` caused problems in some distributions.
